faker_test.go

```diff
@@ -307,8 +307,28 @@ func TestSetDataErrorDataParseTagIntType(t *testing.T) {
 	}{}
 
 	if err := FakeData(temp); err == nil {
-		t.Error("Exptected error Unsupported tag, but got nil")
+		t.Error("Expected error Unsupported tag, but got nil")
+	}
+}
+
+func TestSetNilIfLenIsZero(t *testing.T) {
+	someStruct := SomeStruct{}
+	rand.Seed(0)
+	SetNilIfLenIsZero(true)
+	testRandZero = true
+	if err := FakeData(&someStruct); err != nil {
+		t.Error("Fake data generation has failed")
+	}
+	if someStruct.MapStringString != nil && someStruct.MapStringStruct != nil &&
+		someStruct.MapStringStructPointer != nil {
+		t.Error("Map has to be nil")
+	}
+	if someStruct.Stime != nil && someStruct.SBool != nil {
+		t.Error("Array has to be nil")
 	}
+	rand.Seed(time.Now().UnixNano())
```

Comment:
Hi @olimpias, 
Thanks for the PR. 
I think you can remove from this line of code to belows (329-331)

Reply:
Hi @bxcodec , yes :)